sources/packages/api/src/services/education-program-offering/education-program-offering-import-csv.models.ts

```diff
@@ -0,0 +1,296 @@
+  offeringName: string;
+  /**
+   * Offering study start date.
+   */
+  @IsDateString(undefined, {
+    message: getDateFormatMessage(CSVHeaders.StudyStartDate),
+  })
+  studyStartDate: string;
+  /**
+   * Offering study end date.
+   */
+  @IsDateString(undefined, {
+    message: getDateFormatMessage(CSVHeaders.StudyEndDate),
+  })
+  studyEndDate: string;
+  /**
+   * Actual tuition costs.
+   */
+  @IsNumber(currencyNumberOptions, {
+    message: getCurrencyFormatMessage(CSVHeaders.ActualTuitionCosts),
+  })
+  actualTuitionCosts: number;
+  /**
+   * Program related costs.
+   */
+  @IsNumber(currencyNumberOptions, {
+    message: getCurrencyFormatMessage(CSVHeaders.ProgramRelatedCosts),
+  })
+  programRelatedCosts: number;
+  /**
+   * Mandatory fees.
+   */
+  @IsNumber(currencyNumberOptions, {
+    message: getCurrencyFormatMessage(CSVHeaders.MandatoryFees),
+  })
+  mandatoryFees: number;
+  /**
+   * Exceptional expenses.
+   */
+  @IsNumber(currencyNumberOptions, {
+    message: getCurrencyFormatMessage(CSVHeaders.ExceptionalExpenses),
+  })
+  exceptionalExpenses: number;
+  /**
+   * Offering delivered type.
+   */
+  @IsEnum(OfferingDeliveryOptions, {
+    message: getEnumFormatMessage(
+      CSVHeaders.DeliveredType,
+      OfferingDeliveryOptions,
+    ),
+  })
+  offeringDelivered: OfferingDeliveryOptions;
+  /**
+   * Offering intensity.
+   */
+  @IsEnum(OfferingIntensity, {
+    message: getEnumFormatMessage(
+      CSVHeaders.OfferingIntensity,
+      OfferingIntensity,
+    ),
+  })
+  offeringIntensity: OfferingIntensity;
+  /**
+   * Number of years of study.
+   */
+  @IsNumber()
+  yearOfStudy: number;
+  /**
+   * Show year of study.
+   */
+  @IsEnum(YesNoOptions, {
+    message: getYesNoFormatMessage(CSVHeaders.ShowYearOfStudy),
+  })
+  showYearOfStudy: YesNoOptions;
+  /**
+   * Indicates if the offering has a WIL(work-integrated learning).
+   */
+  @IsEnum(YesNoOptions, {
+    message: getYesNoFormatMessage(CSVHeaders.WILComponent),
+  })
+  WILComponent: YesNoOptions;
+  /**
+   * For an offering that has a WIL(work-integrated learning),
+   * indicates which type.
+   */
+  @ValidateIf(
+    (csvModel: OfferingCSVModel) => csvModel.WILComponent === YesNoOptions.Yes,
+    {
+      message: `${CSVHeaders.WILComponentType} is required when ${CSVHeaders.WILComponent} is set to '${YesNoOptions.Yes}'.`,
+    },
+  )
+  @IsNotEmpty()
+  WILComponentType?: string;
+  /**
+   * User consent to have the offering submitted.
+   */
+  @IsIn([YesNoOptions.Yes], {
+    message: `${CSVHeaders.Consent} must be set to '${YesNoOptions.Yes}'.`,
+  })
+  consent: YesNoOptions;
+  /**
+   * Indicates if the offering should be available for every student to select.
+   */
+  @IsEnum(YesNoOptions, {
+    message: getYesNoFormatMessage(CSVHeaders.PublicOffering),
+  })
+  publicOffering: YesNoOptions;
+  /**
+   * Indicates offering course load.
+   */
+  @IsOptional()
+  @IsNumber()
+  courseLoad?: number;
```

Comment:
can we put a range of numbers here?

Reply:
We can, but it is also part of the `SaveOfferingModel`. I was trying to keep the validation here as dummy as possible and I struggle to draw a line on where we should put one or another.
For instance, the start date before the end date is executed only on `SaveOfferingModel`.
So the idea was to keep these ones as basic as possible.
Someones like yes/no are executed here because they are converted to bool properties later and then the idea is that this model ensures that if will be really a yes/no and not a Yes/No, for instance.